pkg/backend/backend.go

```diff
@@ -120,6 +120,12 @@ type Backend interface {
 	// GetPolicyPack returns a PolicyPack object tied to this backend, or nil if it cannot be found.
 	GetPolicyPack(ctx context.Context, policyPack string, d diag.Sink) (PolicyPack, error)
 
+	// ListPolicyGroups returns all Policy Groups for an organization in this backend, or nil if it cannot be found.
+	ListPolicyGroups(ctx context.Context, orgName string) (apitype.ListPolicyGroupsResponse, error)
```

Comment:
"nil if it cannot be found" actually doesn't match the code, since `apitype.ListPolicyGroupsResponse` cannot have the value `nil`. It might make more sense to return a well-known error, like `var ErrOrganizationNotFound = errors.New("organization not found")` so callers can check for that instead. WDYT?

Reply:
Yeah some better error handling would be good to add 👍

Reply:
Looks like it already an appropriate error -- so ill just update the comments